building Gadgetron with testing requires CMake 3.23 (missing target GTest::gmock)

With CMake 3.17 we get
```
GTEST FOUND: TRUE
...
-- Configuring done
CMake Error at test/CMakeLists.txt:62 (add_executable):
  Target "test_all" links to target "GTest::gmock" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```
https://cmake.org/cmake/help/latest/module/FindGTest.html states that `GTest::gmock` is introduced in CMake 3.23.

We should therefore check CMake version and set `BUILD_TESTING_Gadgetron=OFF` and add a warning if CMake is too old to enhance our user experience.

--- a/SuperBuild/External_Gadgetron.cmake
+++ b/SuperBuild/External_Gadgetron.cmake
@@ -137,6 +137,9 @@ endif()
 
   # Gadgetron only adds tests if (GTEST_FOUND AND ARMADILLO_FOUND)
   if (BUILD_TESTING_${proj})
+    if(CMAKE_VERSION VERSION_LESS "3.23") 
+       message(FATAL_ERROR "You need at least CMake 3.23 to build Gadgetron tests.")
+    endif()
     add_test(NAME ${proj}_TESTS
          COMMAND ${CMAKE_CTEST_COMMAND} -C $<CONFIGURATION> --output-on-failure
          WORKING_DIRECTORY ${${proj}_BINARY_DIR}/test)

